File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        """
        Function to calculate the minimum number of parentheses additions
        required to make the string valid.

        Parameters:
        s (str): A string consisting of '(' and ')'

        Returns:
        int: The minimum number of parentheses to be added
        """

        open_parentheses = 0  # Counter for unbalanced '(' parentheses
        close_parentheses = 0  # Counter for unbalanced ')' parentheses

        # Iterate through each character in the string
        for ch in s:
            if ch == '(':  # If it's an opening parenthesis
                open_parentheses += 1  # Increase the count for '('
            elif ch == ')':
                if open_parentheses > 0:
                    # If there's an unmatched '(', match it with the current ')'
                    open_parentheses -= 1
                else:
                    # If no unmatched '(', increment the ')' counter
                    close_parentheses += 1

        # The total number of unmatched parentheses is the sum of unmatched '(' and ')'
        return open_parentheses + close_parentheses

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.c
int minAddToMakeValid(char* s) {
    int open_parentheses = 0, close_parentheses = 0;
    for (; *s; s++) {
        if (*s == '(') {
            open_parentheses++;
        }
        else if (*s == ')' && open_parentheses > 0) {
            open_parentheses--;
        }
        else {
            close_parentheses++;
        }
    }
    return open_parentheses + close_parentheses;
}